openedx/core/djangoapps/content_libraries/signal_handlers.py

```diff
@@ -203,3 +208,42 @@ def library_collection_entities_changed(sender, instance, action, pk_set, **kwar
 
     for component in components.all():
         _library_collection_component_changed(component, library.library_key)
+
+
+@receiver(XBLOCK_CREATED)
+@receiver(XBLOCK_UPDATED)
+def create_or_update_upstream_downstream_link_handler(**kwargs):
+    """
+    Automatically create or update upstream->downstream link in database.
+    """
+    xblock_info = kwargs.get("xblock_info", None)
+    if not xblock_info or not isinstance(xblock_info, XBlockData):
+        log.error("Received null or incorrect data for event")
+        return
+
+    create_or_update_xblock_upstream_link.delay(str(xblock_info.usage_key))
```

Comment:
I expected that this handler would take care of creating links when importing a course, but it didn't! I only saw a log message for the top-level course block:

> No upstream or upstream_version found for xblock: block-v1:OpenCraft+DemoX+1+type@course+block@course

I had to run the management command to set my imported course links.

Reply:
This should be fixed now.